dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -176,6 +177,32 @@ public WorkflowDAO(SessionFactory factory) {
         return filteredWorkflows.size() == 1 ? Optional.of(filteredWorkflows.get(0)) : Optional.empty();
     }
 
+    /**
+     * Find if a path already exists in the BioWorkflow or Apptool table since we do not want duplicate names between them.
+     * If creating an apptool, check the workflow table and vice versa.
+     *
+     * @param path
+     * @param clazz
+     * @param <T>
+     * @return
+     */
+    public <T extends Workflow> void checkForDuplicateAcrossTables(String path, Class<T> clazz) {
+        final List<Workflow> workflows = findByPath(path, false);
+        final List<T> filteredWorkflows = workflows.stream()
+            .filter(workflow -> workflow.getClass().equals(clazz))
```

Comment:
1. I don't know if we want to handle services. I noticed a couple of comments in the code about handling the fact that workflows and services can have the same path, so I left that as-is.
2.  Since services can have the same path, the filter is needed.

Reply:
> Since services can have the same path, the filter is needed.

If so, could you just filter on the class not being a Service.class?